src/NumberTheory/vinberg.jl

```diff
@@ -0,0 +1,295 @@
+###################################################################
```

Comment:
I find it confusing that the source code for this is in `NumberTheory` but the documentation in `LinearAlgebra`. I would rather keep those in sync. If this belongs to LinearAlgebra, then let's add a `src/LinearAlgebra/` dir?

Then again, should this even go directly into `src`, or perhaps first into `experimental`? There are many things still a bit rough and unclear to me, e.g. the use of matrices to represent lattices.